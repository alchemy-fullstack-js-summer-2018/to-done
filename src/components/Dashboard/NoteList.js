import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

class NoteList extends Component {

  static propTypes = {
    notes: PropTypes.array
  };

  render() {
    const { notes } = this.props;
    if(!notes) return null;

    return (
      <ul>
        {notes.map(note => (
          <Note
            key={note.key}
            note={note}
          />
        ))}
      </ul>
    );
  }
}

export default NoteList;