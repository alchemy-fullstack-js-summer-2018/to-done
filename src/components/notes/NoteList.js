import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

class NoteList extends Component {
  static propTypes = {
    noteList: PropTypes.array,
    onUpdate: PropTypes.func.isRequired,
    onRemove: PropTypes.func.isRequired
  };

  render() {
    const { noteList, onUpdate, onRemove } = this.props;

    return (
      <ul>
        {noteList.map(note => (
          <Note
            key={note.key}
            note={note}
            onUpdate={onUpdate}
            onRemove={onRemove}
          />
        ))}
      </ul>
    );
  }
}
 
export default NoteList;