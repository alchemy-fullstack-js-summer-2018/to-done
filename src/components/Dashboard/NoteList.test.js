import NoteList from './NoteList';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Note List', () => {

  it('renders as expected', () => {
    const notes = [
      { key: '1', title: 'Read', notes: 'Finish egg chapter in the Measure of my Powers', completed: false },
      { key: '2', title: 'Grocery Shopping', notes: 'eggs, butter, sea salt, heirloom tomatoes', completed: false },
      { key: '3', title: 'Wash Car', notes: 'Seriously, wash it already!', completed: false }
    ];

    const wrapper = shallow(<NoteList 
      notes={notes}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});