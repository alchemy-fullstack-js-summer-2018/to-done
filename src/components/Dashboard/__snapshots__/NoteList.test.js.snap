// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note List renders as expected 1`] = `
<ul>
  <Note
    key="1"
    note={
      Object {
        "completed": false,
        "key": "1",
        "notes": "Finish egg chapter in the Measure of my Powers",
        "title": "Read",
      }
    }
  />
  <Note
    key="2"
    note={
      Object {
        "completed": false,
        "key": "2",
        "notes": "eggs, butter, sea salt, heirloom tomatoes",
        "title": "Grocery Shopping",
      }
    }
  />
  <Note
    key="3"
    note={
      Object {
        "completed": false,
        "key": "3",
        "notes": "Seriously, wash it already!",
        "title": "Wash Car",
      }
    }
  />
</ul>
`;
